src/Nethermind/Nethermind.Network/Rlpx/NettyHandshakeHandler.cs

```diff
@@ -180,9 +183,21 @@ protected override void ChannelRead0(IChannelHandlerContext context, IByteBuffer
             if (_logger.IsTrace) _logger.Trace($"Registering {nameof(ZeroPacketSplitter)} for {RemoteId} @ {context.Channel.RemoteAddress}");
             context.Channel.Pipeline.AddLast(new ZeroPacketSplitter(_logManager));
 
-            PacketSender packetSender = new(_serializationService, _logManager, _sendLatency);
-            if (_logger.IsTrace) _logger.Trace($"Registering {nameof(PacketSender)} for {_session.RemoteNodeId} @ {context.Channel.RemoteAddress}");
-            context.Channel.Pipeline.AddLast(packetSender);
+            IPacketSender packetSender;
+            if (_throttlingConfig.ByteLimit == default && _throttlingConfig.Delay == default)
+            {
+                PacketSender sender = new(_serializationService, _logManager, _sendLatency);
+                if (_logger.IsTrace) _logger.Trace($"Registering {nameof(PacketSender)} for {_session.RemoteNodeId} @ {context.Channel.RemoteAddress}");
+                context.Channel.Pipeline.AddLast(sender);
```

Comment:
I think you can simplify this if/else. I mean you could initialize sender only in if and move shared code outside